homeassistant/components/switch/rainmachine.py

```diff
@@ -217,56 +194,62 @@ def name(self) -> str:
         """Return the name of the program."""
         return 'Program: {}'.format(self._entity_json.get('name'))
 
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, HASS-friendly identifier for this entity."""
+        return '{0}_program_{1}'.format(self.device_mac.replace(':', ''),
+                                        self.rainmachine_entity_id)
+
     def turn_off(self, **kwargs) -> None:
         """Turn the program off."""
         import regenmaschine.exceptions as exceptions
 
         try:
-            self._client.programs.stop(self.rainmachine_id)
+            self._client.programs.stop(self.rainmachine_entity_id)
         except exceptions.BrokenAPICall:
             _LOGGER.error('programs.stop currently broken in remote API')
         except exceptions.HTTPError as exc_info:
             _LOGGER.error('Unable to turn off program "%s"',
-                          self.rainmachine_id)
+                          self.rainmachine_entity_id)
             _LOGGER.debug(exc_info)
 
     def turn_on(self, **kwargs) -> None:
         """Turn the program on."""
         import regenmaschine.exceptions as exceptions
 
         try:
-            self._client.programs.start(self.rainmachine_id)
+            self._client.programs.start(self.rainmachine_entity_id)
         except exceptions.BrokenAPICall:
             _LOGGER.error('programs.start currently broken in remote API')
         except exceptions.HTTPError as exc_info:
             _LOGGER.error('Unable to turn on program "%s"',
-                          self.rainmachine_id)
+                          self.rainmachine_entity_id)
             _LOGGER.debug(exc_info)
 
     def _update(self) -> None:
         """Update info for the program."""
         import regenmaschine.exceptions as exceptions
 
         try:
-            self._entity_json = self._client.programs.get(self.rainmachine_id)
+            self._entity_json = self._client.programs.get(
+                self.rainmachine_entity_id)
         except exceptions.HTTPError as exc_info:
             _LOGGER.error('Unable to update info for program "%s"',
-                          self.rainmachine_id)
+                          self.rainmachine_entity_id)
```

Comment:
It's better to refer to the entity ID, since that is how it is known inside Home Assistant